lib/ip_record.js

```diff
@@ -41,15 +41,16 @@ module.exports = function (limits, now) {
     // IPs are limited based on the number of unique email
     // addresses they access.  Take the highest-weighted
     // bad-login event for each email address.
-    var total = 0
-    var seen = {}
+    var weights = {}
     this.lf.forEach(function(info) {
-      var incr = limits.badLoginErrnoWeights[info.e] || 1
-      if (info.u in seen && seen[info.u] < incr) {
-        total -= seen[info.u]
-        seen[info.u] = incr
-      }
-      total += incr
```

Comment:
@rfk - can you add in comments the intent of the code and maybe a short description of the algorithm? The ultra-terse variable names leave me scratching my head, I have no idea what `e` or `u` are.